pkg/agent/openflow/pipeline.go

```diff
@@ -2060,26 +2060,31 @@ func (f *featureNetworkPolicy) dnsPacketInFlow(conjunctionID uint32) binding.Flo
 		Done()
 }
 
-// localProbeFlow generates the flow to forward locally generated packets to ConntrackCommitTable, bypassing ingress
+// localProbeFlows generates the flows to forward locally generated packets to stageConntrack directly, bypassing ingress
 // rules of Network Policies. The packets are sent by kubelet to probe the liveness/readiness of local Pods.
-// On Linux and when OVS kernel datapath is used, it identifies locally generated packets by matching the
-// HostLocalSourceMark, otherwise it matches the source IP. The difference is because:
+// On Linux and when OVS kernel datapath is used, the probe packets are identified by matching the HostLocalSourceMark.
+// On Windows or when OVS userspace (netdev) datapath is used, we need a different approach because:
 // 1. On Windows, kube-proxy userspace mode is used, and currently there is no way to distinguish kubelet generated
 //    traffic from kube-proxy proxied traffic.
 // 2. pkt_mark field is not properly supported for OVS userspace (netdev) datapath.
-// Note that there is a defect in the latter way that NodePort Service access by external clients will be masqueraded as
-// a local gateway IP to bypass Network Policies. See https://github.com/antrea-io/antrea/issues/280.
-// TODO: Fix it after replacing kube-proxy with AntreaProxy.
-func (f *featurePodConnectivity) localProbeFlow(ovsDatapathType ovsconfig.OVSDatapathType) []binding.Flow {
+// when proxyAll is disabled, the probe packets are identified by matching the source IP is the Antrea gateway IP;
+// otherwise, the packets are identified by matching both the Antrea gateway IP and NotServiceCTMark. Note that, when
+// proxyAll is disabled, currently there is no way to distinguish kubelet generated traffic from kube-proxy proxied traffic
+// only by matching the Antrea gateway IP. There is a defect that NodePort Service access by external clients will be
+// masqueraded as the Antrea gateway IP to bypass Network Policies. See https://github.com/antrea-io/antrea/issues/280.
+func (f *featurePodConnectivity) localProbeFlows(ovsDatapathType ovsconfig.OVSDatapathType, proxyAll bool) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	var flows []binding.Flow
 	if runtime.IsWindowsPlatform() || ovsDatapathType == ovsconfig.OVSDatapathNetdev {
 		for ipProtocol, gatewayIP := range f.gatewayIPs {
-			flows = append(flows, IngressSecurityClassifierTable.ofTable.BuildFlow(priorityHigh).
+			fb := IngressSecurityClassifierTable.ofTable.BuildFlow(priorityHigh).
 				Cookie(cookieID).
 				MatchProtocol(ipProtocol).
-				MatchSrcIP(gatewayIP).
-				Action().GotoStage(stageConntrack).
+				MatchSrcIP(gatewayIP)
```

Comment:
From my side, I would prefer to match gw port as the in_port or match the fromSource reg instead of gatewayIP, because ingress stage is after post routing stage, I am not sure if SNAT is taken for no Service IP in future. Matching in_port or source reg is more compatible.

Reply:
For request packets of non-Service connections, FromGatewayRegMark && NotServiceCTMark -> stageConntrack
   -  ✅  client is the local Node, destination is a local Pod
   -  ❎  client is a remote Pod or remote Node, destination is a local Pod (Windows noEncap mode, remote traffic from remote Pod CIDR is sent to local Pod CIDR via Antrea gateway, not uplink)
  I think the packets of the second case should not bypass ingress rules.

Reply:
Make sense.

Reply:
You remind me, we should add state `-rpl+trk` to avoid impact on reply packets witch noEncap mode, since reply packets are possible to goto Metric table for statistics collection.

Reply:
Thanks, updated.